### DISCUSSION TITLE
[v7] Attributes and includes type inferece

### DISCUSSION THREAD

**Discussion by franciscohanna92** [sequelize/sequelize#16121] [Help & Questions] [answered]
Hi there!

Prisma implements type inference of returned objects based on included attributes and relations. For example:
```typescript
const user = await prisma.user.findUnique({
  where: {
    email: 'elsa@prisma.io',
  },
  select: {
    email: true,
    name: true,
  },
  include: {
    profile: true,
    select: {
      avatar: true
    },
  },
})
```

Then the inferred type of `user` is something like:
```typescript
interface User {
   email: string;
   name: string:
   profile: { avatar: string };
} 
```

Is this something that's going to be supported in Sequelize v7?

**Comment by ephys** [comment, marked as answer]
Not in v7, but this sort of type safety is something we're working towards
